borsh-derive/src/lib.rs

```diff
@@ -80,3 +118,38 @@ pub fn borsh_schema(input: TokenStream) -> TokenStream {
         Err(err) => err.to_compile_error(),
     })
 }
+
+#[proc_macro_attribute]
+pub fn borsh(args: TokenStream, input: TokenStream) -> TokenStream {
+    let attribute_args: syn::AttributeArgs = syn::parse_macro_input!(args);
+
+    let attr = MacroAttribute::from_attribute_args(
+        "use_discriminant",
+        attribute_args,
+        syn::AttrStyle::Outer,
+    );
+
+    let mut use_discriminant = false;
+    let mut found = false;
+    let name_values_attributes = attr.into_name_values().unwrap();
+
+    for (name, value) in &name_values_attributes {
+        if name == "use_discriminant" && value.to_string() == "true" {
+            use_discriminant = true;
+            found = true;
```

Comment:
I feel you meant to set `found` to true not only when `value` is `true`, but also when the `value` is `false`, didn't you? Otherwise, there is no role for having two variables that set to `false` at the same time, and set to `true` at the same time.